### PR TITLE
DATAUP-288 Refactor, initial left column container

### PR DESCRIPTION
**Pull request by briehl**
# Description of PR purpose/changes

There were two goals here:
1. Get the first pass at a component for the left column. This is a super hand-wavy thing right now with hard-coded data and state. The main purpose here was to make something that took up space on the DOM, and be an actual component that can start up. No tests yet, and a couple inline styles, I think.

2. Refactor the various other components to use the factory pattern and not classes. We decided to stick with factories for now as that will make it easier to re-use other existing modules and components without either refactoring them or having a crazy switch statement that'll pick and choose how to initialize various components.

Some of this is kinda hurried and slapdash to try to unblock other engineers, but critiques are welcome. If this is a good time to convert inline styles -> scss, and fill out unit tests, then that's cool, too. I don't want to be TOO sloppy. 😄 

# Jira Ticket / Issue #
e.g. https://kbase-jira.atlassian.net/browse/DATAUP-X
- [x] Added the Jira Ticket to the title of the PR e.g. (DATAUP-69 Adds a PR template)

# Testing Instructions
* Details for how to test the PR: 
- [ ] Tests pass in travis and locally 
- [ ] Changes available by spinning up a local narrative and navigating to _X_ to see _Y_

# Dev Checklist:

- [ ] My code follows the guidelines at https://sites.google.com/truss.works/kbasetruss/development
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: kbase-extension/scss/partials/_appCell.scss [hunk 1]
```diff
@@ -150,6 +150,11 @@
     display: none;
   }
 
+  [data-element=advanced-hidden-message] {
```
Comment (ialarmedalien) [inline kbase-extension/scss/partials/_appCell.scss:153]: :sneaky:

^^ imagine there was a great emoji there

Comment (briehl) [reply kbase-extension/scss/partials/_appCell.scss:153]: Yeah, probably not the best answer. I couldn't get a real path to that for some reason and forgot about doing a good BEM name. I'll fix that and be less sneaky.

File: nbextensions/bulkImportCell/bulkImportCell.js [hunk 2]
```diff
@@ -20,26 +22,34 @@ define([
...
+    CategoryPanel
 ) => {
     'use strict';
     const CELL_TYPE = 'app-bulk-import';
 
     const div = html.tag('div');
 
-    class DefaultWidget {
-        constructor() {
+    function DefaultWidget() {
```
Comment (ialarmedalien) [inline nbextensions/bulkImportCell/bulkImportCell.js:35]: (not specific to this line of code!)

It looks like it's just this module that now uses the `class` paradigm -- do you want to convert it over to the factory pattern as part of this PR?

Comment (briehl) [reply nbextensions/bulkImportCell/bulkImportCell.js:35]: I did. I wonder how that got reverted.
Too much branch switching today. Meh.

Comment (briehl) [reply nbextensions/bulkImportCell/bulkImportCell.js:35]: I can do that in this PR if that's helpful. I don't want it to be too annoyingly huge to review.

Comment (ialarmedalien) [reply nbextensions/bulkImportCell/bulkImportCell.js:35]: It's fine by me if it's a bit bigger. It shouldn't make too much difference.

File: kbase-extension/scss/partials/_appCell.scss (hunk 1 above)
Comment (briehl) [reply kbase-extension/scss/partials/_appCell.scss]: Ok, now less sneaky. I BEM'd that and added it to the bulk-import-cell partial.
Also, removed remaining inline styles, I think.

File: nbextensions/bulkImportCell/bulkImportCell.js (hunk 2 above)
Comment (briehl) [reply nbextensions/bulkImportCell/bulkImportCell.js]: It's gonna turn out to be a lot bigger as most lines are touched. I'll set up a new PR specifically for that.

File: nbextensions/bulkImportCell/cellTabs.js [hunk 3]
```diff
@@ -28,41 +34,46 @@ define([
...
          *       }
          */
-        constructor(options) {
-            this.bus = options.bus;
-            this.ui = options.ui;
-            this.tabToggleAction = options.toggleAction;
-            this.controlBarTabs = options.tabs;
-        }
+        let bus = options.bus,
+            ui,
```
Comment (leia-sefkin) [inline nbextensions/bulkImportCell/cellTabs.js:38]: should this be `ui = options.ui`?

**Review by leia-sefkin — APPROVED** [review]
looks good to me. having this consistency is already doing wonders for my ability to understand what the heck is going on here

File: nbextensions/bulkImportCell/cellTabs.js (hunk 3 above)
Comment (briehl) [reply nbextensions/bulkImportCell/cellTabs.js:38]: Nope, that sets up its own. I fixed that in the comments.

**Pull request merged**